### PR TITLE
detect OSD cluster

### PR DESCRIPTION
**Pull request by zhiweiyin318**
related issue:  https://github.com/open-cluster-management/backlog/issues/8841

### CODE REVIEW

File: pkg/klusterlet/clusterinfo/clusterinfo_controller.go [hunk 1]
```diff
@@ -386,6 +390,32 @@ func (r *ClusterInfoReconciler) getCloudVendor() clusterv1beta1.CloudVendorType
...
+	Resource: "subjectpermissions",
+}
+
+const (
+	dedicatedAdmin    = "dedicated-admins"
+	rbacPermissionsNs = "openshift-rbac-permissions"
+)
+
+func (r *ClusterInfoReconciler) isOpenShiftDedicated() bool {
+	_, err := r.ManagedClusterDynamicClient.Resource(subjectpermissionsGVR).
```
Comment (qiujian16) [inline pkg/klusterlet/clusterinfo/clusterinfo_controller.go:407]: could we just check the existence of this api resources? With this, you would need additional rbac

File: pkg/klusterlet/clusterinfo/clusterinfo_controller.go [hunk 2]
```diff
@@ -342,7 +342,11 @@ func (r *ClusterInfoReconciler) getVendor(gitVersion string, isOpenShift bool) (
 	case strings.Contains(gitVersion, string(clusterv1beta1.KubeVendorICP)):
 		kubeVendor = clusterv1beta1.KubeVendorICP
 	case isOpenShift:
-		kubeVendor = clusterv1beta1.KubeVendorOpenShift
+		if r.isOpenShiftDedicated() {
```
Comment (qiujian16) [inline pkg/klusterlet/clusterinfo/clusterinfo_controller.go:345]: this looks strange. why not using a case before this case?

**Comment by qiujian16** [comment]
i think product in claim also needs to be handled.

File: pkg/klusterlet/clusterinfo/clusterinfo_controller.go (hunk 1 above)
Comment (zhiweiyin318) [reply pkg/klusterlet/clusterinfo/clusterinfo_controller.go]: yes. modified to use group name `managed.openshift.io` to detect if the cluster is osd, like using group name `project.openshift.io `to detect if the cluster is OCP.

File: pkg/klusterlet/clusterinfo/clusterinfo_controller.go (hunk 2 above)
Comment (zhiweiyin318) [reply pkg/klusterlet/clusterinfo/clusterinfo_controller.go]: done

**Comment by zhiweiyin318** [comment]
> i think product in claim also needs to be handled.

done

File: pkg/klusterlet/clusterinfo/clusterinfo_controller.go [hunk 3]
```diff
@@ -719,43 +743,53 @@ func (r *ClusterInfoReconciler) GetClusterClaims() ([]*clusterv1alpha1.ClusterCl
...
 	cloudVendor := r.getCloudVendor()
 	gitVersion := strings.ToUpper(kubeVersion)
+
+	// deal with product and platform
 	switch {
 	case strings.Contains(gitVersion, string(clusterv1beta1.KubeVendorIKS)):
-		// IKS
-		platform = string(clusterv1beta1.KubeVendorIKS)
+		// IBM Cloud Platform + IKS
+		platform = IBMPlatformType
```
Comment (qiujian16) [inline pkg/klusterlet/clusterinfo/clusterinfo_controller.go:753]: I do not think we should do this in this release. UI maybe affected.

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**